[CI] Checkstyle license header check fails on Windows

The task `:docs:checkstyleTest` fails regularly on Windows platforms, looks related to #79603.

**Build scan**:
https://gradle-enterprise.elastic.co/s/v4julezaoxpmk
https://gradle-enterprise.elastic.co/s/c2mi2ziflwcnk
https://gradle-enterprise.elastic.co/s/jzykxzzrjaari
and more

**Repro line**:

**Reproduces locally?**:
No (no windows platforms)

**Applicable branches**:
master

**Failure excerpt**:
```
:docs:checkstyleTest FAILED
[ant:checkstyle] [ERROR] C:\Users\jenkins\workspace\platform-support\365\docs\src\test\java\org\elasticsearch\smoketest\DocsClientYamlTestSuiteIT.java:1: Duplicate header javadocs are forbidden [MultipleHeaderJavadoc]
```

--- a/build-tools-internal/src/main/resources/checkstyle.xml
+++ b/build-tools-internal/src/main/resources/checkstyle.xml
@@ -14,7 +14,7 @@
 
   <module name="RegexpMultiline">
     <property name="id" value="MultipleHeaderJavadoc" />
-    <property name="format" value="^\s*\/\*\n(\s\*[A-Za-z0-9 \.\/\;\,\.\-\(\)\\\x{22}\/\:\@\=\'\]\[\_\x{3E}\x{3C}]*\n)+(\s\*\/)\s+package" />
+    <property name="format" value="^\s*\/\*\r?\n(\s\*[A-Za-z0-9 \.\/\;\,\.\-\(\)\\\x{22}\/\:\@\=\'\]\[\_\x{3E}\x{3C}]*\r?\n)+(\s\*\/)\s+package" />
     <property name="fileExtensions" value="java" />
     <property name="minimum" value="1" />
     <property name="maximum" value="1" />

